p5js/NeuralNetwork/NeuralNetwork_basics/sketch.js

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2018 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+/* ===
+ml5 Example
+Image classification using MobileNet and p5.js
+This example uses a callback pattern to create the classifier
+=== */
+let nn;
+
+function setup() {
+  // Options for Neural Network
+  const options = {
+    inputs: 3,
+    outputs: 2,
+    task:'regression',
+    // activationOutput: 'sigmoid',
+    // activationHidden: 'sigmoid'
+  };
+  // Create Neural Network
+  nn = ml5.neuralNetwork(options);
+
+  trainModel();
+
+  // Train the model
+  let trainBtn = createButton('Train Model');
+  trainBtn.position(10, 50);
+  trainBtn.mousePressed(function () {
+    trainModel();
+  });
+
+  // Predict
+  let predictBtn = createButton('Predict');
+  predictBtn.position(10, 70);
+  predictBtn.mousePressed(function () {
+    predict();
+  });
+
+  // Save and download the model
+  let saveBtn = createButton('Save Model');
+  saveBtn.position(10, 90);
+  saveBtn.mousePressed(function () {
+    neuralNetwork.save();
+  });
+
+  // Load the model from local files
+  let loadLocalBtn = createButton('Load the model from local files');
+  loadLocalBtn.position(10, 110);
+  loadLocalBtn.mousePressed(function () {
+    neuralNetwork.load('model/model.json', function () {
+      console.log('Model Loaded!');
+    });
+  });
+
+  // Load model
+  let loadBtn = select('#load');
+  loadBtn.changed(function () {
+    neuralNetwork.load(loadBtn.elt.files, function () {
+      console.log('Model Loaded!');
+    });
+  });
+}
+
+function trainModel() {
+  // Add training data
+  // const training_input = [-0.6, 1, 0.25];
+  // const training_target = [0.3, 0.9];
+  let a,b,c;
+  let training_target;
+  for (let i = 0; i < 500; i++) {
+    if(i % 2){
+       a = Math.random(0,0.16);
+       b = Math.random(0.16,0.32);
+       c = Math.random(0.32,0.5);
+       training_target = [0,0]
+    }else {
+      a = Math.random(0.5,0.66);
+      b = Math.random(0.66,0.82);
+      c = Math.random(0.82,1);
+      training_target = [1,1]
+    }
+    
+    const training_input = [a, b, c];
+    // const training_target = [0, 1];
+
+    nn.data.addData({
+      input0: training_input[0],
+      input1: training_input[1],
+      input2: training_input[2],
+    }, 
+    {
+      output0: training_target[0],
+      output1: training_target[1],
+    });
+  }
+
+  const trainingOptions = {
+    epochs: 100,
+    batchSize: 12
+  }
+  // Train
+  nn.data.shuffle();
```

Comment:
I believe you addressed this in a comment somewhere @joeyklee but just noting that training fails if you do not call `shuffle()` or `normalize()`. The tensors are not created/shaped properly without a call to one of these functions. While one could make the argument you should always  normalize, the training should technically be possible without shuffling or normalizing.

I also think that while it's nice to keep `shuffle()` as a helper function, the conventional time to shuffle is during training itself after each epoch. So this is something that our `train()` function could do natively, with an optional options flag to turn that off if you wanted to for any reason.

Reply:
Re shuffle:
Yes, I've removed the shuffle step in the examples since this is now included in the training. 

Re: training
Noted! Yes, right now the training is called on `data.normalizedData.inputs` and `data.normalizedData.targets` as a result, the call to `normalize()` for now is necessary. I will think about how to change this. Thanks!